### PR TITLE
21 06 16

### PR DESCRIPTION
**Pull request by lindajohnstone**

### CODE REVIEW

File: ConwaysGameOfLife.Tests/CellShould.cs [hunk 1]
```diff
@@ -42,16 +42,44 @@ public void WhenInitialized_SetLocation(int x, int y)
         }
 
         [Theory]
-        [InlineData(CellState.Dead, 0, 1, CellState.Dead, 0, 1, true)]
-        [InlineData(CellState.Dead, 0, 1, CellState.Alive, 0, 1, false)]
-        public void GivenTwoCells_CellStateAndLocationAreEqual(CellState state1, int x1, int y1,  CellState state2, int x2, int y2, bool expected) 
+        [InlineData(CellState.Dead, 0, 1, CellState.Dead, 0, 1)]
+        [InlineData(CellState.Alive, 0, 1, CellState.Alive, 0, 1)]
+        public void GivenTwoCells_CellStateAndLocationAreEqual(CellState state1, int x1, int y1,  CellState state2, int x2, int y2)
```
Comment (kedoodle) [inline ConwaysGameOfLife.Tests/CellShould.cs:47]: good, can we go one step further given that we're now passing the exact same thing to the constructors of each of the two cells?

also a naming suggestion
```suggestion
        [InlineData(CellState.Dead, 0, 1)]
        [InlineData(CellState.Alive, 0, 1)]
        public void BeEqual_GivenSameCellStateAndLocation(CellState state, int x, int y) 
```

Comment (indianbread) [inline ConwaysGameOfLife.Tests/CellShould.cs:47]:
```suggestion
        public void BeEquivalent_GivenSameStateAndLocation(CellState state1, int x1, int y1,  CellState state2, int x2, int y2) 

```

File: ConwaysGameOfLife.Tests/CellShould.cs [hunk 2]
```diff
@@ -42,16 +42,44 @@ public void WhenInitialized_SetLocation(int x, int y)
...
         [Theory]
-        [InlineData(CellState.Dead, 0, 1, CellState.Dead, 0, 1, true)]
-        [InlineData(CellState.Dead, 0, 1, CellState.Alive, 0, 1, false)]
-        public void GivenTwoCells_CellStateAndLocationAreEqual(CellState state1, int x1, int y1,  CellState state2, int x2, int y2, bool expected) 
+        [InlineData(CellState.Dead, 0, 1, CellState.Dead, 0, 1)]
+        [InlineData(CellState.Alive, 0, 1, CellState.Alive, 0, 1)]
+        public void GivenTwoCells_CellStateAndLocationAreEqual(CellState state1, int x1, int y1,  CellState state2, int x2, int y2) 
         {
             var one = new Cell(state1, x1, y1);
             var two = new Cell(state2, x2, y2);
```
Comment (kedoodle) [inline ConwaysGameOfLife.Tests/CellShould.cs:50]: worth having cell in the names perhaps
```suggestion
            var cell1 = new Cell(state1, x1, y1);
            var cell2 = new Cell(state2, x2, y2);
```

File: ConwaysGameOfLife.Tests/CellShould.cs [hunk 3]
```diff
@@ -42,16 +42,44 @@ public void WhenInitialized_SetLocation(int x, int y)
...
-            Assert.Equal(expected, result);
+            Assert.True(result);
+        }
+
+        [Theory]
+        [MemberData(nameof(GetCellFromDataGenerator))]
+        public void GivenTwoCells_CellStateAndLocation_AreEqual(CellState state1, int x1, int y1, CellState state2, int x2, int y2)
+        {
+            var one = new Cell(state1, x1, y1);
+            var two = new Cell(state2, x2, y2);
```
Comment (kedoodle) [inline ConwaysGameOfLife.Tests/CellShould.cs:62]: same as above comments re: naming and params

File: ConwaysGameOfLife.Tests/CellShould.cs [hunk 4]
```diff
@@ -42,16 +42,44 @@ public void WhenInitialized_SetLocation(int x, int y)
...
             var two = new Cell(state2, x2, y2);
 
             var result = UniverseHelper.CellsAreEqual(one, two);
 
-            Assert.Equal(expected, result);
+            Assert.True(result);
+        }
+
+        [Theory]
+        [MemberData(nameof(GetCellFromDataGenerator))]
```
Comment (indianbread) [inline ConwaysGameOfLife.Tests/CellShould.cs:58]: nice!

File: ConwaysGameOfLife.Tests/CellShould.cs [hunk 5]
```diff
@@ -42,16 +42,44 @@ public void WhenInitialized_SetLocation(int x, int y)
...
+        public void GivenTwoCells_CellStateAndLocation_AreEqual(CellState state1, int x1, int y1, CellState state2, int x2, int y2)
```
Comment (indianbread) [inline ConwaysGameOfLife.Tests/CellShould.cs:59]: rename this as it's not testing for equivalence

File: ConwaysGameOfLife.Tests/CellShould.cs [hunk 6]
```diff
@@ -42,16 +42,44 @@ public void WhenInitialized_SetLocation(int x, int y)
...
+        {
+            var one = new Cell(state1, x1, y1);
+            var two = new Cell(state2, x2, y2);
+
+            var result = UniverseHelper.CellsAreEqual(one, two);
+
+            Assert.False(result);
+        }
+
+        public static IEnumerable<object[]> GetCellFromDataGenerator()
```
Comment (indianbread) [inline ConwaysGameOfLife.Tests/CellShould.cs:69]: Alternative: Use TheoryData to allow you to use Cell objects - we can pair on this

File: ConwaysGameOfLife.Tests/ValidatorShould.cs [hunk 7]
```diff
@@ -10,11 +10,11 @@ public ValidatorShould()
             _validator = new Validator();
         }
 
-        [Fact]
-        public void AcceptUniverse_WhenInputIsValid()
+        [Theory]
+        [InlineData("3,3")]
+        [InlineData("10,3")]
+        public void ReturnTrue_GivenValidUniverseInputputIsValid(string input)
```
Comment (indianbread) [inline ConwaysGameOfLife.Tests/ValidatorShould.cs:16]:
```suggestion
        public void ReturnTrue_GivenValidUniverseInput(string input)

```

**Review by indianbread — APPROVED** [review]

File: ConwaysGameOfLife.Tests/ValidatorShould.cs (hunk 7 above)
Comment (kedoodle) [inline ConwaysGameOfLife.Tests/ValidatorShould.cs:16]:
```suggestion
        public void ReturnTrue_GivenValidUniverseInput(string input)
```

File: ConwaysGameOfLife.Tests/ValidatorShould.cs [hunk 8]
```diff
@@ -38,7 +39,7 @@ public void NotAcceptUniverse_WhenInputIsInvalid(string input)
         [Theory]
         [InlineData("0,0", 3, 3)]
         [InlineData("2,1", 3, 3)]
```
Comment (kedoodle) [inline ConwaysGameOfLife.Tests/ValidatorShould.cs:41]: Can we expand tests on the grid width and length? You've already done the hard work - just add a couple more cases with different dimension grids.

File: ConwaysGameOfLife.Tests/ValidatorShould.cs [hunk 9]
```diff
@@ -55,7 +56,7 @@ public void AcceptLocation_WhenInputIsValid(string input, int gridWidth, int gri
         [InlineData("a,b", 3, 3)]
         [InlineData("-1,4", 3, 3)]
         [InlineData("4,-1", 3, 3)]
```
Comment (kedoodle) [inline ConwaysGameOfLife.Tests/ValidatorShould.cs:58]: Same here, can we get some different size grid widths and lengths? What happens if we pass in negative value widths and lengths? (Not that we would)

**Review by kedoodle — APPROVED** [review]
awesome, not much to comment on today. would like to see the generator class fleshed out a bit next. maybe showcase the ability to end the game by pressing q?

File: ConwaysGameOfLife.Tests/CellShould.cs (hunk 6 above)
Comment (lindajohnstone) [reply ConwaysGameOfLife.Tests/CellShould.cs:69]: Yes please

**Pull request merged**